explore_server:
  ros__parameters:
    frequency: 15.0
    goal_aliasing: 2.0
    retry_count: 20
    nav2_goal_timeout_sec: 50 #secs
    use_traversability: false

explore_client:
  ros__parameters:
    costmap_resolution: 0.05 # same as bounded_explore_layer
  
polygon_point_publisher:
  ros__parameters:
    square_half_side_length: 15.0

path_follower_node:
  ros__parameters:
    translation_speed: 0.5
    rotation_speed: 1.0
    local_goal_distance: 0.4 # min distance for local goal
    resolution: 0.05 # resolution of global costmap
    
frontier_selection_node:
  ros__parameters:
    exploration_mode: ours
    counter: 031150
    alpha: 0.26
    beta: 0.49
    planner_allow_unknown: true
    frontier_detect_radius: 0.5

localization_pose_sub_node:
  ros__parameters:
    exploration_mode: ours
    counter: 031150

explore_costmap:
  explore_costmap:
    ros__parameters:
      footprint: "[ [0.48, 0.38], [0.48, -0.38], [-0.48, -0.38], [ -0.48, 0.38 ] ]"
      update_frequency: 1.0
      publish_frequency: 1.0
      global_frame: map
      robot_base_frame: base_link
      use_sim_time: True
      resolution: 0.05
      track_unknown_space: true
      # robot_radius: 0.22
      # plugins: ["traversability_layer", "bounded_explore_layer"]
      # plugins: ["bounded_explore_layer"]
      # plugins: ["static_layer"]
      plugins: ["static_layer", "bounded_explore_layer"]
      static_layer:
        plugin: "nav2_costmap_2d::StaticLayer"
        map_subscribe_transient_local: True
      bounded_explore_layer:
        plugin: "frontier_exploration/BoundedExploreLayer"
        enabled: true
        resize_to_boundary: true
        frontier_travel_point: closest
        exploration_mode: ours
        explore_clear_space: false
        min_frontier_cluster_size: 1
        use_traversability: false
      always_send_full_costmap: false